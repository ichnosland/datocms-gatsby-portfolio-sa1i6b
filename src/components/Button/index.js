/**
*
* Button
*
*/

import styled, { css } from 'styled-components';
//import { lighten, darken, transparentize } from 'polished';
import media from 'style/mediainjector';
import { colore, transparent } from 'style/color';
import Svg from '../Svg';

export const ButtonStyle = css`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  font-family: ${(props) => props.theme.fonts.buttons.fontFamily};
  font-weight: ${(props) => props.theme.fonts.buttons.fontWeight};
  ${({ full }) => full && 'width: 100% !important;'};
  ${({ half }) => half && 'width: 50% !important;'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  text-align: center;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding ? props.padding : '0.4em  1.4em'};
  border: ${(props) => props.border ? props.border : '1px solid transparent'};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : props.theme.brand};
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.buttons};
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : props.theme.brand))};
    color: ${(props) => transparentize(0.5, (props.color ? props.color : colore.ui.contrast))};
    pointer-events: none;
    svg {
        opacity: .3;
      }
  }
  ${({ round }) => round && css`
    padding: 0;
    width: ${(props) => props.width};
    height: ${(props) => props.width};
    border-radius: 100%;
  `}
  ${({ actioncolor }) => actioncolor && css`
    background-color: ${colore.actions[actioncolor]};
    &:disabled {
      background-color: ${lighten(0.3, colore.actions[actioncolor])};
      color: ${lighten(0.1, colore.actions[actioncolor])};
    }
  `}
  ${({ admin }) => admin && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${colore.ui.neutralBg};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
  `}
  ${({ outline }) => outline && css`
    color: ${(props) => props.theme.brand};
    border: 1px solid ${(props) => props.theme.brand};
    background-color: ${colore.ui.contrast};
    &:disabled {
      background-color: ${darken(0.1, colore.ui.contrast)};
      color: ${(props) => lighten(0.4, props.theme.brand)};
    }
    &:active {
      background-color: ${darken(0.1, colore.ui.contrast)};
      border-color: ${darken(0.1, colore.ui.neutralBg)};
    }
    svg {
      fill: ${(props) => props.fill ? props.fill : props.theme.brand};
    }
  `}
  ${({ bone }) => bone && css`
    color: ${colore.ui.contrast};
    border: 2px solid ${colore.ui.contrast};
    background-color: ${transparent};
    &:hover,
    &:active {
      background-color: rgba(0,0,0,.1);
    }
  `}
  &:active,
  &:focus {
    outline: none;
  }
  ${(props) => props.standard && css`
    min-width: 180px;
    max-width: 320px;
  `};
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${media.iphone45`
    ${({ standard }) => standard && 'min-width: 140px;'}
    padding: ${(props) => props.padding ? props.padding : '0.4em  0.6em;'};
    svg {
      margin: 0 6px 0 0;
    }
  `}
`;

export const Button = styled.button`
  ${ButtonStyle}
`;

Button.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const SubmitButton = styled.input`
  ${ButtonStyle}
`;

SubmitButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
    fonts: {
      buttons: {
        fontFamily: 'inherit',
        fontWeight: 'inherit',
      },
    },
  },
  actioncolor: '',
};

export const ActionButtonStyle = css`
  font-size: 21px;
  padding: 21px 30px;
  background-color: ${(props) => props.bgcolor};
  border: ${(props) => props.border ? props.border : '2px solid white'};
  color: ${(props) => props.color ? props.color : colore.ui.contrast};
  font-weight: 600;
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  ${({ sectionbutton }) => sectionbutton && css`
    min-width: 50%;
    font-size: 18px;
    padding: 14px 30px;
    ${media.lt667`
      width: 100%;
    `}
  ${media.lt480`
    font-size: 16px;
    `}
  `}
  &:disabled {
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : colore.landing.green))};
    color: ${(props) => transparentize(0.5, (props.bgcolor ? props.bgcolor : colore.ui.contrast))};
  }
`;

export const ActionButton = styled(Button)`
  ${ActionButtonStyle};
`;

export const Icon = styled(Svg)`
  width: ${(props) => props.size ? props.size : '18px'};
  height: ${(props) => props.size ? props.size : '18px'};
  margin: ${(props) => props.margin};
  fill: ${(props) => props.fill ? props.fill : colore.ui.contrast};
  ${({ right }) => right && css`
    margin: 0 -6px 0 6px;
  `}
  ${({ left }) => left && css`
    margin: 0 6px 0 -6px;
  `}
  ${({ brandColor }) => brandColor && css`
    fill: ${(props) => props.theme.brand};
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

Icon.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const NoStyleButton = css`
  appearance: none;
  background: none;
  border-top: none;
  border-left: none;
  border-right: none;
  border-bottom: none;
  margin: ${(props) => props.margin};
  padding: ${(props) => props.padding};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const GhostButton = styled.button`
  ${NoStyleButton}
  display: inline-flex;
  align-items: center;
  justify-content: center;
  svg {
    fill: ${(props) => props.theme.brand};
  }
  ${(props) => props.fill && css`
    svg {
      fill: ${props.fill};
    }
  `}
  &.mediumUp {
    @media (max-width: 667px) {
      display: none;
    }
  }
`;

GhostButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const TextButton = styled.button`
  color: ${(props) => props.color ? props.color : props.theme.brand};
  appearance: none;
  border: none;
  background: none;
  &:active,
  &:focus {
    outline: none;
  }
  &:hover {
    ${media.desktop`
      color: ${(props) => props.theme.darken};
    `}
  }
`;

TextButton.defaultProps = {
  theme: {
    brand: '#00BBEF',
    darken: '#1588AA',
  },
};
