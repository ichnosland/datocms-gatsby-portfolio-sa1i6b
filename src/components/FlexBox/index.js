/**
*
* FlexBox
*
*/
import styled, { css } from 'styled-components';

import Div from '../Div';
import media from 'style/mediainjector';

export const FlexBox = styled(Div)`
  display: flex;
  flex-direction: ${(props) => props.direction ? props.direction : 'row'};
  justify-content: ${(props) => props.justifyContent ? props.justifyContent : 'center'};
  align-items: ${(props) => props.alignItems ? props.alignItems : 'center'};
  align-content: ${(props) => props.alignContent ? props.alignContent : 'center'};
  flex-wrap: ${(props) => props.wrap ? props.wrap : 'nowrap'};
  ${media.print`
   ${({ noPrint }) => noPrint && css`
       display:none;
   `}
 `}
`;

export const FlexChild = styled(FlexBox)`
  order: ${(props) => props.order};
  flex-grow: ${(props) => props.flexGrow};
  flex-basis: ${(props) => props.flexBasis};
  align-self: ${(props) => props.alignSelf};
`;

export const FlexWrap = styled(FlexBox)`
  position: relative;
  flex-direction: column;
  justify-content: space-between;
  min-height: 100%;
  height: 100%;
  ${media.desktop`
    &:before {
      content: '';
      display: block;
    }
  `}
  ${media.mobile`
    display: block;
  `}
`;

export default FlexBox;
