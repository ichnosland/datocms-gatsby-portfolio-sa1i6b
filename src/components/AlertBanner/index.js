/**
*
* AlertBanner
*
*/

import styled, { css } from 'styled-components';
import { lighten, darken } from 'polished';
import FlexBox from '../FlexBox';
import media from 'style/mediainjector';
import { colore } from 'style/color';

export const BannerStyle = css`
  ${({ tag }) => tag && css`display: inline-flex;`};
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  text-align: center;
  margin: ${(props) => props.margin ? props.margin : '0'};
  border: ${(props) => props.border ? `1px solid ${props.theme.brand}` : `1px solid ${colore.ui.neutralBg}`};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : colore.ui.contrast};
  ${({ actioncolor }) => actioncolor && css`
    border: 1px solid ${colore.actions[actioncolor]};
    background-color: ${lighten(0.4, colore.actions[actioncolor])};
    color: ${darken(0.2, colore.actions[actioncolor])};
    svg {
      fill: ${darken(0.2, colore.actions[actioncolor])};
    }
  `}
  ${({ shadow }) => shadow && css`
    border: none;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
  border-radius: ${(props) => props.radius ? props.radius : props.theme.radius.general};
  &:active,
  &:focus {
    outline: none;
  }
`;

export const AlertBanner = styled(FlexBox)`
  ${BannerStyle}
  padding: ${(props) => props.padding ? props.padding : '0.6em  1.6em'};
  svg {
    fill: ${({ color, actioncolor }) => {
    if (color) return color;
    if (actioncolor) return actioncolor;

    return colore.ui.txt;
  }};
  }
`;

AlertBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const InfoBanner = styled(AlertBanner)`
  padding: ${(props) => props.padding ? props.padding : '0.6em  1em'};
  ${media.lt667`
    flex-direction: column;
    svg {
      margin: 10px 0;
    }
  `}
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '0.6em'};
  `}
`;

InfoBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export const IconBanner = styled(FlexBox)`
  ${BannerStyle}
  align-self: stretch;
  align-items: center;
  padding: ${(props) => props.padding ? props.padding : '9px 16px'};
  div {
    margin-right: 12px;
    align-items: center;
    &:last-child {
      margin-right: 0;
    }
  }
  ${media.iphone45`
    padding: ${(props) => props.padding ? props.padding : '7px 12px'};
  `}
`;

IconBanner.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      general: '6px',
    },
  },
  actioncolor: '',
};

export default AlertBanner;
