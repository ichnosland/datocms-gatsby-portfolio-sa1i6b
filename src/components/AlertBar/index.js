/**
*
* AlertBar
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import styled, { css } from 'styled-components';

import Div from '../Div';
import FlexBox from '../FlexBox';
import Svg from '../Svg';
import { IconBtn } from '../TopBar';
import { H3 } from '../Heading';
import media from 'style/mediainjector';
import { colore } from 'style/color';
import icon from 'icons/globals';
import { ContainerStyle } from '../Container';

export const AlertWrap = styled(Div)`
  position: absolute;
  top: 0;
  left: 0;
  color: ${colore.ui.contrast};
  width: 100%;
  height: 56px;
  padding: 0 20px;
  background-color: ${(props) => props.theme.brand};
  ${({ giusto }) => giusto && css`background-color: ${colore.actions.okay};`}
  ${({ sbagliato }) => sbagliato && css`background-color: ${colore.actions.error};`}
  ${({ aiuto }) => aiuto && css`background-color: ${colore.actions.help};`}
  z-index: 8;
  transition: all 0.5s;
  ${media.desktop`
    border-top-left-radius: ${(props) => props.versione ? '3px' : '20px'};
    border-top-right-radius: ${(props) => props.versione ? '3px' : '20px'};
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

AlertWrap.defaulProps = {
  theme: {
    brand: '#00BBEF',
  },
};

export const BarContainer = styled(FlexBox)`
  ${ContainerStyle}
  height: 100%;
  ${(props) => props.helpFx && css`
    justify-content: space-between;
    &::after {
      content: "";
      display: inline;
    }
  `}
  h3 {
    color: ${colore.ui.contrast};
  }
`;

class AlertBar extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      giusto,
      sbagliato,
      aiuto,
      versione,
      show,
      helpFx,
    } = this.props;
    return (
      <AlertWrap
        giusto={giusto}
        sbagliato={sbagliato}
        aiuto={aiuto}
        show={show}
        versione={versione}
      >
        <BarContainer helpFx={helpFx}>
          {helpFx &&
            <IconBtn onClick={helpFx}>
              <Svg {...icon.topBarFeedback} />
            </IconBtn>
          }
          <H3>
            {giusto &&
              'Risposta corretta!'
            }
            {sbagliato &&
              'Hai sbagliato!'
            }
            {aiuto &&
              'Aiuto'
            }
          </H3>
        </BarContainer>
      </AlertWrap>
    );
  }
}

AlertBar.propTypes = {
  giusto: PropTypes.bool,
  sbagliato: PropTypes.bool,
  aiuto: PropTypes.bool,
  versione: PropTypes.bool,
  show: PropTypes.bool.isRequired,
  helpFx: PropTypes.func,
};

export default AlertBar;
