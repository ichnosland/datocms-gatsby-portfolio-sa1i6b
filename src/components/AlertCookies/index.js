/**
*
* AlertCookies
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Div from 'components/Div';
import P from 'components/Paragraph';
import { Button } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const AlertCookiesWrap = styled(Div)`
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 0 20px 20px;
  z-index: 1;
  ${media.lt667`
    padding: 0 10px 30px;
  `}
  ${({ show }) => !show && css`
    display: none;
  `}
`;

export const CookiesBanner = styled(Div)`
  color: ${(props) => props.color ? props.color : colore.ui.txt};
  max-width: 860px;
  margin: 0 auto;
  font-size: 14px;
  border: ${(props) => props.border};
  background-color: ${(props) => props.bgcolor ? props.bgcolor : 'rgba(255,255,255,.95)'};
  border-radius: 4px;
  border: 1px solid rgba(0,0,0,.15);
  box-shadow: 0 2px 1px rgba(0,0,0,.25);
  p {
    padding: 20px;
    margin: 0;
    &:last-child {
      padding: 0 20px 10px;
      text-align: right;
      ${media.lt480`
        padding: 0 20px 20px;
        text-align: center;
      `}
    }
  }
  ${media.lt667`
    font-size: 12px;
  `}
`;

class AlertCookies extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { show, color, border, bgcolor, onClickFunction } = this.props;
    return (
      <AlertCookiesWrap show={show}>
        <CookiesBanner
          color={color}
          border={border}
          bgcolor={bgcolor}
        >
          <P>Al fine di offrire il miglior servizio possibile questo sito utilizza cookies tecnici e di terze parti per funzionalità quali l&apos;accesso tramite social network e/o la visualizzazione di media. Se non acconsenti all&apos;utilizzo dei cookie di terze parti, alcune di queste funzionalità potrebbero essere non disponibili. Chiudendo questo banner o proseguendo con la navigazione acconsentirai al loro impiego. Per saperne di pi&ugrave; o negare il consenso a tutti o ad alcuni cookie puoi consultare <a href="https://maieuticallabs.it/privacy" target="_blank"><span>l&apos;informativa sulla privacy</span></a>.</P>
          <P><Button actioncolor="hint" standard={1} onClick={onClickFunction}>Chiudi</Button></P>
        </CookiesBanner>
      </AlertCookiesWrap>
    );
  }
}

AlertCookies.propTypes = {
  color: PropTypes.string,
  border: PropTypes.string,
  bgcolor: PropTypes.string,
  show: PropTypes.bool.isRequired,
  onClickFunction: PropTypes.func,
};

export default AlertCookies;
