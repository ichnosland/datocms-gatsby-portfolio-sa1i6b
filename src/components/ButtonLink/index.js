/**
*
* ButtonLink
*
*/
import styled, { css } from 'styled-components';

import { Link } from 'react-router-dom';
import { ButtonStyle, ActionButtonStyle } from 'components/Button';
import { lighten, transparentize } from 'polished';
import { colore } from 'style/color';

export const ButtonLink = styled(Link)`
  ${ButtonStyle}
  text-decoration: none;
  ${({ inactive }) => inactive && css`
    background-color: ${(props) => lighten(0.3, (props.bgcolor ? props.bgcolor : props.theme.brand))};
    color: ${(props) => transparentize(0.5, (props.color ? props.color : colore.ui.contrast))};
    pointer-events: none;
  `}
`;

ButtonLink.defaultProps = {
  theme: {
    brand: '#00BBEF',
    radius: {
      buttons: '6px',
    },
  },
  actioncolor: '',
};

export const ActionButtonLink = styled(ButtonLink)`
  ${ActionButtonStyle};
`;
