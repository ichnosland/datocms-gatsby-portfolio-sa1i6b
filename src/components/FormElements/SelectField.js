/**
*
* SelectField
*
*/

import React from 'react';
import PropTypes from 'prop-types';

import Svg from 'components/Svg';
import icon from 'icons/globals';
import { SelectBox, SelectChevron, SelectMenu } from 'components/Select';
import Spinner from 'components/Spinner';
import { FormMessage } from './index';


class SelectField extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      input,
      children,
      meta: { asyncValidating, touched, error },
      margin,
      padding,
      radius,
      bgColor,
      color,
      optioncolor,
      border,
      arrowColor,
      className,
    } = this.props;
    return ([
      touched && error && <FormMessage landing key="select_message">{error}</FormMessage>,
      <SelectBox
        key="select_box"
        margin={margin}
        padding={padding}
        radius={radius}
        bgColor={bgColor}
        color={color}
        border={border}
        className={className}
      >
        <SelectMenu
          color={color}
          optioncolor={optioncolor}
          {...input}
        >
          {children}
        </SelectMenu>
        <SelectChevron arrowColor={arrowColor}>
          <Svg {...icon.chevronDown} />
        </SelectChevron>
      </SelectBox>,
      asyncValidating && <Spinner size="small" key="select_spinner" />,
    ]);
  }
}

SelectField.propTypes = {
  label: PropTypes.string,
  name: PropTypes.string,
  id: PropTypes.string,
  input: PropTypes.object,
  margin: PropTypes.string,
  padding: PropTypes.string,
  radius: PropTypes.string,
  bgColor: PropTypes.string,
  color: PropTypes.string,
  optioncolor: PropTypes.string,
  border: PropTypes.string,
  arrowColor: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.any,
  meta: PropTypes.object,
};

export default SelectField;
