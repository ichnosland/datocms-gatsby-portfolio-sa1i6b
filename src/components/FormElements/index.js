/**
*
* FormElements
*
*/

import styled, { css } from 'styled-components';

import Section from 'components/Section';
import Div from 'components/Div';
import { Button, Icon } from 'components/Button';
import { colore } from 'style/color';
import media from 'style/mediainjector';

export const BasicInputStyle = css`
  display: block;
  box-sizing: border-box;
  color: ${(props) => props.theme.brand};
  width: 100%;
  text-align: ${(props) => props.align ? props.align : 'left'};
  appearance: none;
  padding: 6px 13px;
  border: 1px solid ${(props) => props.theme.light};
  background-color: ${colore.ui.contrast};
  ${({ note }) => note && css`
    color: ${colore.ui.txt};
    background-color: ${colore.ui.note};
    border-color: ${colore.ui.noteBorder};
  `}
  &::placeholder {
    position: relative;
    color: ${(props) => props.theme.subtle};
    ${({ note }) => note && css`
      color: ${colore.ui.noteBorder};
    `}
    opacity: 1;
    overflow: visible;
  }
  &:active,
  &:focus {
    outline: none;
  }
`;

BasicInputStyle.defaultProps = {
  theme: {
    brand: '#00BBEF',
    subtle: '#B4E1FF',
    light: '#C9E4FF',
    radius: {
      searchBox: '100px',
    },
  },
};

export const FormSection = styled(Section)`
  height: 100%;
  flex-direction: column;
  justify-content: space-between;
  &::before {
    content: "";
    display: inline;
  }
`;

export const CardForm = styled.form`
  width: 280px;
  margin: 0 auto;
  text-align: center;
  ${media.lt480`
    width: 240px;
  `}
`;

export const FormElement = styled(Div)`
  position: relative;
  margin: ${(props) => props.margin ? props.margin : '0 auto 18px'};
  ${(props) => props.errore && css`
    margin-bottom: 9px;
  }`
}`;


export const FormMessage = styled.span`
  display: inline-block;
  width: 100%;
  font-size: 15px;
  color: ${(props) => props.color ? props.color : colore.actions.error};
  padding: 6px 9px;
  margin-top: 9px;
  border: 1px solid ${colore.ui.neutralBg};
  background-color: ${colore.ui.contrast};
  border-radius: ${(props) => props.theme.radius.general};
  ${(props) => props.landing && css`
    font-size: smaller;
    padding: 2px 12px;
    border-radius: 60px;
    margin-bottom: 3px;
    box-shadow: 0 0 4px 1px rgba(0,0,0,0.20);
  `}
`;

FormMessage.defaultProps = {
  theme: {
    radius: {
      general: '6px',
    },
  },
};

export const FacebookButton = styled(Button)`
  background-color: ${colore.social.facebook};
  border-radius: 3px;
`;

export const FacebookIcon = styled(Icon)`
  top: -2px;
  fill: #fff;
`;
