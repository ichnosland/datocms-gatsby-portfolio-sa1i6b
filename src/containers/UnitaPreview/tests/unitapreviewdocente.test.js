import React from 'react';
import { shallow } from 'enzyme';

import { FormElement } from 'components/FormElements';
import { ButtonLink } from 'components/ButtonLink';
import { Button } from 'components/Button';
import { FlexChild } from 'components/FlexBox';
import SelectField from 'components/FormElements/SelectField';

import {
  UnitaPreviewDocenteView,
  SelectBtn,
  UnitButton,
  UnitBlock,
} from '../UnitaPreviewDocente';


const mockProps = {
  buttonAssegnaIcon: 'bell',
  buttonAssegnaLabel: 'label button assegnazione',
  buttonAssegnaDisabled: false,
  unitaId: 555,
  invalid: false,
  lezioni: [{
    optionLabel: 'lezione1',
    optionKey: 0,
  }, {
    optionLabel: 'lezione2',
    optionKey: 1,
  }, {
    optionLabel: 'lezione3',
    optionKey: 2,
  }],
  tempoEsecuzione: 40,
  eseguiUnitaFunction: () => { },
  buttonAssegnaFunction: () => { },
};

describe('<UnitaPreviewDocenteView />', () => {
  it('visualizzo elementi di gestione della classe', () => {
    const renderedComponent = shallow(
      <UnitaPreviewDocenteView {...mockProps} />
    );
    expect(renderedComponent.find('form').length).toBe(1);
    expect(renderedComponent.find(FormElement).length).toBe(1);
    expect(renderedComponent.find(ButtonLink).length).toBe(0);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<SelectBtn />', () => {
  it('should render its css w/ props theme', () => {
    const theme = {
      brand: 'red',
    };
    const renderedComponent = shallow(
      <SelectBtn theme={theme}>
        <option value="value 1">Value1</option>
        <option value="Value 2">Value 2</option>
      </SelectBtn>
    );
    expect(renderedComponent.find(SelectField).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<UnitButton />', () => {
  it('should render its css w/ props theme if disabled', () => {
    const theme = {
      brand: 'green',
    };
    const renderedComponent = shallow(
      <UnitButton theme={theme} disabled>
        <span>Button</span>
      </UnitButton>
    );
    expect(renderedComponent.find(Button).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<UnitBlock />', () => {
  it('should render its last-child css w/ props theme', () => {
    const theme = {
      light: 'yellow',
    };
    const renderedComponent = shallow(
      <UnitBlock theme={theme} disabled>
        <div>div 1</div>
        <div>div 2</div>
        <div>div 3</div>
      </UnitBlock>
    );
    expect(renderedComponent.find(FlexChild).length).toBe(1);
    expect(renderedComponent).toMatchSnapshot();
  });
});
