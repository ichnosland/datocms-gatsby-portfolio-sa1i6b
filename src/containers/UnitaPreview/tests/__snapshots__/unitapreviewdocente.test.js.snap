// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectBtn /> should render its css w/ props theme 1`] = `
.c0 {
  position: relative;
}

.c0 select {
  color: red;
  padding: 7px 30px 7px 25px;
}

.c0 svg {
  fill: red;
}

@media (max-width:64em) {
  .c0 select {
    padding: 9px 30px 9px 25px;
  }
}

<SelectField
  className="c0"
  theme={
    Object {
      "brand": "red",
    }
  }
>
  <option
    value="value 1"
  >
    Value1
  </option>
  <option
    value="Value 2"
  >
    Value 2
  </option>
</SelectField>
`;

exports[`<UnitBlock /> should render its last-child css w/ props theme 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0:last-child {
  margin-bottom: 30px;
}

@media (min-width:64.0625em) {
  .c0 {
    min-height: 180px;
    width: 50%;
    max-width: 50%;
  }

  .c0:last-child {
    border-left: 1px solid yellow;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
    margin-bottom: 0;
  }
}

<FlexBox__FlexChild
  className="c0"
  disabled={true}
  theme={
    Object {
      "light": "yellow",
    }
  }
>
  <div>
    div 1
  </div>
  <div>
    div 2
  </div>
  <div>
    div 3
  </div>
</FlexBox__FlexChild>
`;

exports[`<UnitButton /> should render its css w/ props theme if disabled 1`] = `
.c0 {
  width: 96px;
  height: 96px;
  padding: 0;
  text-align: center;
  background-color: rgb(245,194,35);
  box-shadow: 0 0 0 6px rgba(0,0,0,0.1);
}

.c0 svg {
  width: 32px;
  height: 39px;
  margin: 0;
  fill: #fff;
}

.c0:disabled {
  background-color: green;
  box-shadow: 0 0 0 6px rgba(255,255,255,0.5);
}

<Button
  actioncolor=""
  className="c0"
  disabled={true}
  theme={
    Object {
      "brand": "green",
    }
  }
>
  <span>
    Button
  </span>
</Button>
`;

exports[`<UnitaPreviewDocenteView /> visualizzo elementi di gestione della classe 1`] = `
<form
  onSubmit={[Function]}
>
  <UnitaPreviewDocente__UnitBlockWrap>
    <UnitaPreviewDocente__UnitBlock
      theme={
        Object {
          "light": "#C9E4FF",
        }
      }
    >
      <FormElements__FormElement>
        <Paragraph
          margin="0 auto 18px"
          theme={
            Object {
              "brand": "#00BBEF",
              "darken": "#1588AA",
            }
          }
        >
          Seleziona una lezione per provarla
        </Paragraph>
        <StyledButtonGroup__MiniButtonGroup
          theme={
            Object {
              "radius": Object {
                "buttons": "6px",
              },
            }
          }
        >
          <Field
            arrowColor="rgb(96, 96, 96)"
            bgColor="#fff"
            border="1px solid #fff"
            color="rgb(96, 96, 96)"
            component={[Function]}
            name="lezione"
            radius="30px"
          >
            <option
              id="lezione_0"
              key="lezione_0"
              value={0}
            >
              lezione1
            </option>
            <option
              id="lezione_1"
              key="lezione_1"
              value={1}
            >
              lezione2
            </option>
            <option
              id="lezione_2"
              key="lezione_2"
              value={2}
            >
              lezione3
            </option>
          </Field>
          <UnitaPreviewDocente__StretchButton
            actioncolor="okay"
            border="1px solid white"
            disabled={false}
            padding="8px 25px"
            type="submit"
          >
            <Svg
              fill="#fff"
              height="18px"
              path={
                <path
                  d="M.657 8.25h14.917L8.558 1.183C8.286.91 8.286.47 8.563.202c.272-.27.708-.267.975 0L17.8 8.528c.268.27.268.708-.004.98L9.533 17.8c-.135.135-.313.2-.487.2-.18 0-.352-.065-.488-.2-.272-.273-.272-.71 0-.98L15.6 9.75H.658C.29 9.75 0 9.413 0 9s.29-.75.657-.75"
                  fillRule="evenodd"
                />
              }
              viewBox="0 0 18 18"
              width="18px"
            />
          </UnitaPreviewDocente__StretchButton>
        </StyledButtonGroup__MiniButtonGroup>
      </FormElements__FormElement>
    </UnitaPreviewDocente__UnitBlock>
    <UnitaPreviewDocente__UnitBlock
      theme={
        Object {
          "light": "#C9E4FF",
        }
      }
    >
      <Paragraph
        margin="0 auto 18px"
        theme={
          Object {
            "brand": "#00BBEF",
            "darken": "#1588AA",
          }
        }
      >
        label button assegnazione
      </Paragraph>
      <UnitaPreviewDocente__UnitButton
        disabled={false}
        onClick={[Function]}
      >
        <Svg
          height="18px"
          path={
            <path
              d="M28 15.744V24l3 5H5l3-5v-8.256c.215-5.13 3.621-9.239 8.04-10.325V4.256C16.04 3 16.72 2 18.02 2 19.32 2 20 3 20 4.256V5.42c4.42 1.086 7.785 5.196 8 10.325zM18 34a3 3 0 0 1-3-3h6a3 3 0 0 1-3 3z"
              fillRule="nonzero"
            />
          }
          viewBox="0 0 36 36"
          width="18px"
        />
      </UnitaPreviewDocente__UnitButton>
    </UnitaPreviewDocente__UnitBlock>
  </UnitaPreviewDocente__UnitBlockWrap>
  <UnitaPreviewDocente__LastP>
    Tempo di esecuzione stimato:
    <br />
    circa 
    40
     minuti
  </UnitaPreviewDocente__LastP>
</form>
`;
